chart/templates/configmap-env.yaml

```diff
@@ -31,8 +31,12 @@ data:
   MALLOC_ARENA_MAX: "2"
   NODE_ENV: "production"
   RAILS_ENV: "production"
+  {{- if .Values.redis.enabled }}
   REDIS_HOST: {{ template "mastodon.redis.fullname" . }}-master
-  REDIS_PORT: "6379"
+  {{- else }}
+  REDIS_HOST: {{ .Values.redis.hostname }}
```

Comment:
Since this is required if redis is disabled it'd be nice to add validation for it, perhaps something like: [required "When the redis chart is disabled .Values.redis.hostname is required" .Values.redis.hostname](https://helm.sh/docs/howto/charts_tips_and_tricks/#know-your-template-functions)?

Reply:
We could also do it the same way that it is being done for postgres
```yaml
  {{- if .Values.redis.enabled }}
  REDIS_HOST: {{ template "mastodon.redis.fullname" . }}-master
  REDIS_PORT: "6379"
  {{- else }}
  REDIS_HOST: {{ .Values.redis.hostname }}
  REDIS_PORT: {{ .Values.redis.port | default "6379" | quote }}
  {{- end }}
```
This way we avoid making the `port` a required setting and we keep retro compatibility